scripts/route_check.py

```diff
@@ -493,6 +522,53 @@ def filter_out_standalone_tunnel_routes(routes):
     return updated_routes
 
 
+def check_frr_pending_routes():
+    """
+    Check FRR routes for offload flag presence by executing "show ip route json"
+    Returns a list of routes that have no offload flag.
+    """
+
+    missed_rt = []
+
+    retries = FRR_CHECK_RETRIES
+    for i in range(retries):
+        missed_rt = []
+        frr_routes = get_frr_routes()
+
+        for _, entries in frr_routes.items():
+            for entry in entries:
+                if entry['protocol'] != 'bgp':
+                    continue
+
+                # TODO: Also handle VRF routes. Currently this script does not check for VRF routes so it would be incorrect for us
+                # to assume they are installed in ASIC_DB, so we don't handle them.
+                if entry['vrfName'] != 'default':
+                    continue
+
+                if not entry.get('offloaded', False):
+                    missed_rt.append(entry)
+
+        if not missed_rt:
+            break
+
+        time.sleep(FRR_WAIT_TIME)
+
+    return missed_rt
+
+
+def mitigate_installed_not_offloaded_frr_routes(missed_frr_rt, rt_appl):
```

Comment:
could you explain how mitigate work here?

Reply:
@StormLiangMS Extended doc string with more information about mitigate function

Reply:
@stepanblyschak , mitigation cannot be done from route_check.py. This is only a tool for comparing ASIC/APP DBs.

Reply:
@prsunny This was a requirement from MSFT. Please clarify what is the suggestion then.

Reply:
@StormLiangMS , discussed offline with @stepanblyschak and plan to have this temporary mitigation until a better solution is available. Kindly review.